api/src/wfl/service/datarepo.clj

```diff
@@ -73,25 +77,24 @@
   (ingest
    "ingest"
    dataset-id
-   {:format                "json"
-    :load_tag              "string"
-    :max_bad_records       0
-    :path                  path
-    :table                 table}))
+   {:format          "json"
+    :load_tag        (new-load-tag)
+    :max_bad_records 0
+    :path            path
+    :table           table}))
 
 (defn poll-job
-  "Return result for JOB-ID in ENVIRONMENT when it stops running."
-  [job-id]
-  (let [get-result #(-> (repository "jobs/" job-id "/result")
-                        (http/get {:headers (auth/get-service-account-header)})
-                        util/response-body-json)
-        running?   #(-> (repository "jobs/" job-id)
-                        (http/get {:headers (auth/get-service-account-header)})
-                        util/response-body-json
-                        :job_status
-                        #{"running"})]
-    (while (running?) (.sleep TimeUnit/SECONDS 1))
-    (get-result)))
+  "Poll the job with `job-id` every `seconds` [default: 5] and return its
+   result."
+  ([job-id seconds]
+   (let [result   #(get-repository-json "jobs" job-id "result")
+         running? #(-> (get-repository-json "jobs" job-id)
+                       :job_status
+                       #{"running"})]
+     (while (running?) (.sleep TimeUnit/SECONDS seconds))
+     (result)))
+  ([job-id]
```

Comment:
Clojure q - is a multi-arity function the best / only way to provide default argument(s)?  What about the case where you might wish to allow multiple default arguments?

Reply:
It's certainly one way, not sure if it's the best way. You can simulate named arguments by taking key-value pairs. Here's one way to implement that, not sure if it's the best:
```clojure
(let [defaults {:option1 value1 :option2 value2}]
  (defn my-named-args-function
    [required & options]
      (let [inputs (apply assoc defaults options)]
        (use inputs))))
```